### PR TITLE
<feat> : Audit alerts from BPFLSM enforcer

### LINKED ISSUE #1477: Using Enforcer for Audit Policy alerts
## Feature Request

**Short Description**

Currently, we have alerts from enforcer for Block events only, we should be able to use enforcer for audit events also 

**Describe the solution you'd like**

Extension of issue #1277 

**Checklist:**
- [x] use BPFLSM, enforcer for Default posture audit alerts  https://github.com/kubearmor/KubeArmor/pull/1492
- [ ] use BPFLSM enforcer for Audit Policy alerts  https://github.com/kubearmor/KubeArmor/pull/1644

### PR DESCRIPTION
**Pull request by Aryan-sharma11**
**Purpose of PR?**:

fixes #1477

**Does this PR introduce a breaking change?**
No
**If the changes in this PR are manually verified, list down the scenarios covered:**:
That there are no Default Posture audit alerts with `enforcer: eBPF monitor`  in  BPFLSM

**Additional information for reviewer?** :
_Mention if this PR is part of any design or a continuation of previous PRs_
https://github.com/kubearmor/KubeArmor/pull/1259
**Checklist:**
- [ ] Bug fix. Fixes #<issue number>
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] PR Title follows the convention of  `<type>(<scope>): <subject>`
- [ ] Commit has unit tests
- [ ] Commit has integration tests

### CODE REVIEW

File: KubeArmor/enforcer/bpflsm/enforcer.go [hunk 1]
```diff
@@ -318,15 +319,24 @@ func (be *BPFEnforcer) TraceEvents() {
 			log.Operation = "File"
 			log.Resource = string(bytes.Trim(event.Data.Path[:], "\x00"))
 			log.Enforcer = "BPFLSM"
-			log.Result = "Permission denied"
+			if event.Retval == -EPERM {
+				log.Result = "Permission denied"
+			} else {
+				log.Result = "Passed"
+			}
+
```
Comment (daemon1024) [inline KubeArmor/enforcer/bpflsm/enforcer.go]: We should refactor this out of switch case. it's being repeated

File: KubeArmor/enforcer/bpflsm/enforcer.go [hunk 2]
```diff
@@ -318,15 +319,24 @@ func (be *BPFEnforcer) TraceEvents() {
...
 			var sockProtocol int32
 			sockProtocol = int32(event.Data.Path[1])
 			log.Operation = "Network"
 			log.Enforcer = "BPFLSM"
-			log.Result = "Permission denied"
+			if event.Retval == -EPERM {
+				log.Result = "Permission denied"
+			} else {
+				log.Result = "Passed"
+			}
```
Comment (daemon1024) [inline KubeArmor/enforcer/bpflsm/enforcer.go]:
```suggestion
			if event.Retval >=0 {
			        log.Result = "Passed"
			} else {
				log.Result = "Permission denied"
			}
```

File: KubeArmor/enforcer/bpflsm/enforcer.go [hunk 3]
```diff
@@ -389,15 +403,7 @@ func (be *BPFEnforcer) DestroyBPFEnforcer() error {
...
 
-	for _, link := range be.Probes {
-		if link == nil {
-			continue
-		}
-		if err := link.Close(); err != nil {
-			be.Logger.Err(err.Error())
-			errBPFCleanUp = true
-		}
-	}
```
Comment (daemon1024) [inline KubeArmor/enforcer/bpflsm/enforcer.go]: Why are we doing this?

**Review by daemon1024 — CHANGES_REQUESTED** [review]

File: KubeArmor/enforcer/bpflsm/enforcer.go (hunk 3 above)
Comment (Aryan-sharma11) [reply KubeArmor/enforcer/bpflsm/enforcer.go]: Sorry, It was a change I did during testing.

**Comment by rksharma95** [comment]
i tested it on an eks cluster and it looks good. :+1:

File: KubeArmor/enforcer/bpflsm/enforcer.go [hunk 4]
```diff
@@ -341,9 +340,13 @@ func (be *BPFEnforcer) TraceEvents() {
...
 			log.Resource = string(bytes.Trim(event.Data.Path[:], "\x00"))
 			log.Enforcer = "BPFLSM"
-			log.Result = "Permission denied"
 			log.Data = "lsm=" + mon.GetSyscallName(int32(event.EventID))
 		}
+		if event.Retval == -EPERM {
+			log.Result = "Permission denied"
+		} else {
+			log.Result = "Passed"
+		}
```
Comment (daemon1024) [inline KubeArmor/enforcer/bpflsm/enforcer.go]:
```suggestion

		if event.Retval >= 0 {
			log.Result = "Passed"
		} else {
			log.Result = "Permission denied"
		}
```

File: KubeArmor/enforcer/bpflsm/enforcer.go [hunk 5]
```diff
@@ -341,9 +340,13 @@ func (be *BPFEnforcer) TraceEvents() {
 			log.Source = string(bytes.Trim(event.Data.Source[:], "\x00"))
 			log.Resource = string(bytes.Trim(event.Data.Path[:], "\x00"))
 			log.Enforcer = "BPFLSM"
```
Comment (daemon1024) [inline KubeArmor/enforcer/bpflsm/enforcer.go]: We should move this line to after Switch Case as well. And see if we can handle anything else in a common way to deduplicate things.

**Review by daemon1024 — APPROVED** [review]

File: KubeArmor/enforcer/bpflsm/enforcer.go (hunk 4 above)
Comment (daemon1024) [reply KubeArmor/enforcer/bpflsm/enforcer.go]: This will eliminate the need for EPERM declaration

**Review by rksharma95 — APPROVED** [review]
LGTM 👍

**Review by Ankurk99 — APPROVED** [review]
LGTM!

**Comment by Ankurk99** [comment]
fixes: https://github.com/kubearmor/KubeArmor/pull/1473

**Comment by Ankurk99** [comment]
It won't close https://github.com/kubearmor/KubeArmor/pull/1473 as that's for [google.golang.org/grpc](https://github.com/grpc/grpc-go)

**Pull request merged**